tube/etl/indexers/base/logic.py

```diff
@@ -0,0 +1,81 @@
+from pyspark.sql.functions import col, array_contains, expr
+
+
+def create_filter_from_json(json_filter):
+    if json_filter is None:
+        return None
+    if json_filter.get("op").lower() in ["and", "or"]:
+        return CompoundLogic(json_filter)
+    else:
+        return SimpleLogic(json_filter)
+
+
+def build_filter_query(current_logic, new_fields):
+    if type(current_logic) == CompoundLogic:
+        return f" {current_logic.op} ".join(
+            [build_filter_query(logic, new_fields) for logic in current_logic.logics]
+        )
+    if type(current_logic) == SimpleLogic:
+        if current_logic.op == "in" and type(current_logic.value) == list:
+            s_value = ",".join(s for s in current_logic.value)
+            value_in_query = f"({s_value})"
+        elif type(current_logic.value) == str:
+            value_in_query = f"'{current_logic.value}'"
+        else:
+            value_in_query = current_logic.value
+        if current_logic.op == "contains":
+            new_fields.append(
+                expr(f"array_contains({current_logic.prop}, {value_in_query})")
+            )
```

Comment:
Is there any place in documentation where we can mention what all `op`s can be used? Also, I'm trying to figure out what is the difference between `in` and `contains` here. Could you let me know?

Reply:
`in` is used for the normal list, while `contains` is used for nested data

Reply:
Thanks, I still think it would be better if we can add some documentation about `filter` and different operations in [tube/docs/configuration_file.md](https://github.com/uc-cdis/tube/blob/master/docs/configuration_file.md)